### PR TITLE
Add aws-jwk-authorizer support

### PR DESCRIPTION
**Pull request by jmortlock**
### WHY

Start of a common JS library for auth, currently adds support for aws-sdk authorizers and sets up the build / publish pipeline

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+    "nock": "^11.7.0",
+    "jest": "^24.9.0"
+  },
+  "dependencies": {
+    "jsonwebtoken": "^8.5.1",
+    "jwk-to-pem": "^2.0.2",
+    "node-fetch": "^2.6.0"
+  },
+  "engines": {
+    "node": "^8 || ^10 || ^12"
```
Comment (HellRok) [inline package.json]: Shouldn't this only be 10 and 12 since we don't test against 8?

File: __test__/authorizer.test.js [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+    const event = {
+      authorizationToken: "Bearer test-auth-invalid"
+    };
+
+    await expect(authorizeStagingToken(event, {})).rejects.toThrow(
+      "Unauthorized"
+    );
+  });
+
+  test("Validates correct token", async () => {
```
Comment (HellRok) [inline __test__/authorizer.test.js]: I wouldn't mind seeing one more test for an out of date token

File: package.json (hunk 1 above)
Comment (jmortlock) [reply package.json]: Resolved, lazy copy paste

File: __test__/authorizer.test.js (hunk 2 above)
Comment (jmortlock) [reply __test__/authorizer.test.js]: Added a test for expired tokens as well

File: src/authorizer.js [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+          e: k.e
+        };
+        const pem = jwkToPem(jwkArray);
+
+        // Verify the token:
+        const decoded = jwk.verify(token, pem, { issuer });
+        return generatePolicy(decoded.sub, "Allow", event.methodArn);
+      })
+      .catch(err => {
+        console.log(err);
```
[the remark this answers is not in the document]
Comment (jmortlock) [reply src/authorizer.js]: No to log from aws lambda you use console.log, console.error etc.

**Pull request merged**